GoogleSignIn/Tests/Unit/GIDProfileDataFetcherTest.m

```diff
@@ -0,0 +1,150 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+// Test module imports
+@import GoogleSignIn;
+
+#import "GoogleSignIn/Sources/GIDHTTPFetcher/Implementations/Fakes/GIDFakeHTTPFetcher.h"
+#import "GoogleSignIn/Sources/GIDProfileDataFetcher/Implementations/GIDProfileDataFetcher.h"
+#import "GoogleSignIn/Tests/Unit/OIDAuthState+Testing.h"
+#import "GoogleSignIn/Tests/Unit/OIDTokenResponse+Testing.h"
+
+static NSString *const kErrorDomain = @"ERROR_DOMAIN";
+static NSInteger const kErrorCode = 400;
+
+@interface GIDProfileDataFetcherTest : XCTestCase
+@end
+
+@implementation GIDProfileDataFetcherTest {
+  GIDProfileDataFetcher *_profileDataFetcher;
+  GIDFakeHTTPFetcher *_httpFetcher;
+}
+
+- (void)setUp {
+  [super setUp];
+  _httpFetcher = [[GIDFakeHTTPFetcher alloc] init];
+  _profileDataFetcher = [[GIDProfileDataFetcher alloc] initWithHTTPFetcher:_httpFetcher];
+}
+
+- (void)testFetchProfileData_outOfAuthState_success {
```

Comment:
We should also test the "no ID token" and "non-fat ID token" cases as well.  This would provide coverage of the successful UserInfo endpoint fetch path as well.

Reply:
Done.